src/activities/bargame/bargame.js

```diff
@@ -0,0 +1,196 @@
+/* GCompris - bargame.js
+ *
+ * Copyright (C) 2016 UTKARSH TIWARI <iamutkarshtiwari@kde.org >
+ *
+ * Authors:
+ *   Yves Combe (GTK+ version)
+ *   UTKARSH TIWARI <iamutkarshtiwari@kde.org > (Qt Quick port)
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import GCompris 1.0 as GCompris
+.import QtQuick 2.0 as Quick
+
+var numberBalls = [[1, 4], [2, 6], [3, 6]]
```

Comment:
For me, numberBalls and boardSize are related to the mode so I would have in the qml something like:
property var modes = [ "easy": {"minBalls": 1, "maxBalls": 4, "boardSize": 15 }, "medium": ...  ] (not sure if syntaxicly correct but the aim would be to have everything related to one mode in the same array).
If it is the same for the others, then you can do the same